src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DRSSubmitToArchiveCommand.java

```diff
@@ -0,0 +1,383 @@
+                                    .build();
+                            client = HttpClients.custom().setSSLContext(sslContext)
+                                    .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE).build();
+                        } catch (KeyManagementException e) {
+                            // TODO Auto-generated catch block
+                            e.printStackTrace();
+                        } catch (NoSuchAlgorithmException e) {
+                            // TODO Auto-generated catch block
+                            e.printStackTrace();
+                        } catch (KeyStoreException e) {
+                            // TODO Auto-generated catch block
+                            e.printStackTrace();
+                        }
+                    }
+                    if (client == null) {
+                        client = HttpClients.createDefault();
+                    }
+                    HttpPost ingestPost;
+                    try {
+                        ingestPost = new HttpPost();
+                        ingestPost.setURI(new URI(drsConfigObject.getString(DRS_ENDPOINT)));
+
+                        byte[] encoded = Base64.getDecoder().decode(System.getProperty(RSA_KEY).replaceAll("[\\r\\n]", ""));
+
+                        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+                        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(encoded);
+                        RSAPrivateKey privKey = (RSAPrivateKey) keyFactory.generatePrivate(keySpec);
+                        //RSAPublicKey publicKey;
+                        /*
+                         * If public key is needed: encoded = Base64.decodeBase64(publicKeyPEM);
+                         * 
+                         * KeyFactory keyFactory = KeyFactory.getInstance("RS256"); X509EncodedKeySpec
+                         * keySpec = new X509EncodedKeySpec(encoded); return (RSAPublicKey)
+                         * keyFactory.generatePublic(keySpec); RSAPublicKey publicKey = new
+                         * RSAPublicKey(System.getProperty(RS256_KEY));
+                         */
+                        Algorithm algorithmRSA = Algorithm.RSA256(null, privKey);
+                        
+                        String body = drsConfigString;
+                        String jwtString = createJWTString(algorithmRSA, BrandingUtil.getInstallationBrandName(), body, jwtTimeout);
+                        logger.info("JWT: " + jwtString);
+
+                        ingestPost.setHeader("Authorization", "Bearer " + jwtString);
+
+                        logger.info("Body: " + body);
+                        ingestPost.setEntity(new StringEntity(body, "utf-8"));
+                        ingestPost.setHeader("Content-Type", "application/json");
+
+                        try (CloseableHttpResponse response = client.execute(ingestPost)) {
```

Comment:
Dumb question. Does Dataverse PUSH the bags to DRS? Or does Dataverse send some config to DRS and then DRS DOES a pull of the bags? I've only played with the local bag archiver.

Reply:
It's a two-step push - Bag/datacite.xml to S3 (hence `extends S3SubmitToArchiveCommand`) and then send a separate message to the DRS Ingest Mgmt service. The latter is what triggers DRS to start processing and includes the DRS-specific info (such as billing codes) - thus keeping the Bag/datacite.xml files sent the same as with any other system.

Reply:
Got it. Thanks!